websites/O/Oculus/presence.ts

```diff
@@ -0,0 +1,207 @@
+const presence = new Presence({
+  clientId: "837833278777065503"
+}),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+/**
+* Replaces "-" with " ", optional first letter of every word in uppercase (the first letter will be always uppercase)
+* @param input Input replacing "-" with " "; First letter always uppercase
+* @param everyFirstLetterUppercase If true, first letter of each word is capitalized 
+* @returns {string}
+* @example // (false) "terms-of-service" -> "Terms of service"; (true) "rift-s" -> "Rift S"
+*/
+function splitOnDashes(input: string, everyFirstLetterUppercase = false) {
+  return input.split("-").map((s, i) => i === 0 || everyFirstLetterUppercase ? s.charAt(0).toUpperCase() + s.slice(1) : s).join(" ");
+}
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "oculus-logo3",
+    startTimestamp: browsingStamp
+  },
+    hostName = document.location.hostname.replace("www.", ""),
+    path = window.location.pathname.split("/").slice(1),
+    showButtons: boolean = await presence.getSetting("buttons"),
+    showTimestamp: boolean = await presence.getSetting("timestamp");
+
+  switch (hostName) {
+    // Support pages
+    case "support.oculus.com": {
+      presenceData.details = "Viewing support page:";
+      if (path[0] === "" || !path[0]) presenceData.state = "Home";
+      else {
+          // Gets the biggest title of page
+          const article = document.querySelector("h1")?.textContent || document.querySelector("h2")?.textContent || document.querySelector("h4")?.textContent;
+          if (article) {
+            presenceData.state = article.length > 128 ? `${article.slice(0, 125)}...` : article;
+            presenceData.buttons = [{
+              label: "Open article",
+              url: `https://${hostName}/${path[0]}`
+            }];
+          } else 
+            presenceData.state = "Unknown article";
+          
+      }
+      break;
+    }
+
+    // Main pages
+    case "oculus.com": {
+      // Hompage
+      if (path[0] === "" || !path[0]) presenceData.details = "Viewing home page";
+      else {
+          switch (path[0]) {
+            case "legal": {
+              presenceData.smallImageKey = "reading";
+              presenceData.smallImageText = "Reading";
+              presenceData.details = splitOnDashes(path[1]);
+              break;
+            }
+
+            // Specific headsets
+            case "quest":
+            case "quest-2":
+            case "go":
+            case "rift":
+            case "rift-s": {
+              presenceData.details = "Viewing the page:";
+              if (path[1] === "accessories") presenceData.state = `Accessories for ${splitOnDashes(path[0], true)}`;
+              else presenceData.state = splitOnDashes(path[0], true);
```

Comment:
that's not prettified by prettier right? xD
![image](https://user-images.githubusercontent.com/9348108/120108081-dbe3e200-c163-11eb-9e70-bf0c592d180f.png)

Reply:
Actually I prettified it, but nothing happened. After a downgrade of the prettier extension, it worked!
937158c